orchagent/fdborch.cpp

```diff
@@ -482,26 +693,69 @@ bool FdbOrch::addFdbEntry(const FdbEntry& entry, const string& port_name, const
     attr.value.s32 = SAI_PACKET_ACTION_FORWARD;
     attrs.push_back(attr);
 
-    if (m_entries.count(entry) != 0) // we already have such entries
+    if(macUpdate)
     {
-        removeFdbEntry(entry);
-    }
+        /* delete and re-add fdb entry instead of update, 
+         * as entry may age out in HW/ASIC_DB before
+         * update, causing the update request to fail.
+         */
+        SWSS_LOG_NOTICE("MAC-Update FDB %s in %s on from-%s:to-%s from-%s:to-%s", entry.mac.to_string().c_str(), vlan.m_alias.c_str(), oldPort.m_alias.c_str(), port_name.c_str(), oldType.c_str(), type.c_str());
+        status = sai_fdb_api->remove_fdb_entry(&fdb_entry);
+        if (status != SAI_STATUS_SUCCESS)
+        { 
+            SWSS_LOG_INFO("FdbOrch RemoveFDBEntry: Failed to remove FDB entry. mac=%s, bv_id=0x%lx",
+                           entry.mac.to_string().c_str(), entry.bv_id);
+        }
+        status = sai_fdb_api->create_fdb_entry(&fdb_entry, (uint32_t)attrs.size(), attrs.data());
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to create %s FDB %s on %s, rv:%d",
+                    type.c_str(), entry.mac.to_string().c_str(), port_name.c_str(), status);
+            return false;
+        }
 
-    sai_status_t status = sai_fdb_api->create_fdb_entry(&fdb_entry, (uint32_t)attrs.size(), attrs.data());
-    if (status != SAI_STATUS_SUCCESS)
+        if (oldPort.m_bridge_port_id != port.m_bridge_port_id)
+        {
+            oldPort.m_fdb_count--;
+            m_portsOrch->setPort(oldPort.m_alias, oldPort);
+            port.m_fdb_count++;
+            m_portsOrch->setPort(port.m_alias, port);
+        }
```

Comment:
If call removeFdbEntry(), these codes are not needed.

Reply:
If the port in the old fdb entry was different, then the fdb count needs to be decremented for the old port and incremented for the new port.

Reply:
If call removeFdbEntry(), the fdb count is already decremented for the old port in removeFdbEntry().

Reply:
removeFdbEntry() will cause the reference count in sairedis to decrement, but here we decrement the count we maintain in local port structure.